packages/cli/src/commands/migrate/migrations/skeleton-3.ts

```diff
@@ -0,0 +1,369 @@
+			return false;
+		}
+		if (argOneExpression.getName() !== 'resolve') {
+			return false;
+		}
+		const callArgs = argOne.getArguments();
+		if (callArgs.length !== 1) {
+			return false;
+		}
+		const firstCallArg = callArgs[0];
+		if (!firstCallArg.isKind(SyntaxKind.StringLiteral) || firstCallArg.getLiteralText() !== '@skeletonlabs/skeleton') {
+			return false;
+		}
+		if (!argTwo.isKind(SyntaxKind.StringLiteral) || argTwo.getLiteralText() !== '../**/*.{html,js,svelte,ts}') {
+			return false;
+		}
+		return true;
+	}
+
+	const configNode = getDefaultExportObject(sourceFile);
+	if (configNode) {
+		const contentProperty = configNode.getProperty('content');
+		if (contentProperty && ts.isPropertyAssignment(contentProperty.compilerNode)) {
+			const propertyAssignment = contentProperty.asKindOrThrow(SyntaxKind.PropertyAssignment);
+			const initializer = propertyAssignment.getInitializer();
+			if (initializer && ts.isArrayLiteralExpression(initializer.compilerNode)) {
+				const arrayLiteral = initializer.asKindOrThrow(SyntaxKind.ArrayLiteralExpression);
+				const elements = arrayLiteral.getElements();
+				for (const element of elements) {
+					if (isJoinedContent(element)) {
+						element.replaceWithText("contentPath(import.meta.url, 'svelte')");
+					}
+				}
+			}
+		}
+	}
+
+	// Imports
+	sourceFile.getImportDeclarations().forEach((importDeclaration) => {
+		const specifier = importDeclaration.getModuleSpecifierValue();
+		if (specifier === '@skeletonlabs/tw-plugin') {
+			importDeclaration.addNamedImport('contentPath');
+			importDeclaration.setModuleSpecifier('@skeletonlabs/skeleton/plugin');
+		}
+		if (['path', 'node:path'].includes(specifier)) {
+			const namedImports = importDeclaration.getNamedImports();
+			const joinImport = namedImports.find((imp) => imp.getName() === 'join');
+			if (joinImport) {
+				if (namedImports.length === 1 && !isModuleUsed(sourceFile, 'join')) {
+					importDeclaration.remove();
+				} else {
+					joinImport.remove();
+				}
+			}
+		}
+	});
+
+	return sourceFile.getFullText();
+}
+
+function migrateClasses(code: string) {
+	return CLASS_REGEXES.reduce((result, migration) => {
+		return result.replace(migration.find, migration.replace);
+	}, code);
+}
+
+function migrateSvelteCode(code: string) {
+	code = migrateClasses(code);
+	// TODO: Generate AST (svelte/compiler)
+	// TODO: Update imports
+	// TODO: Update components
+	return code;
+}
+
+function migrateModuleCode(code: string) {
+	code = migrateClasses(code);
+	// TODO: Generate AST (ts-morph)
+	// TODO: Update imports
+	return code;
+}
+
+function migrateProject(cwd = process.cwd()) {
+	const migration = spinner();
+	migration.start('Starting migration!');
+
+	// package.json
+	const packageMatcher = 'package.json';
+	const packagePath = fg.sync(`./${packageMatcher}`, { cwd })[0];
+	if (!packagePath) {
+		cancel(`"${packageMatcher}" not found, please migrate from the root of your project.`);
```

Comment:
Not sure if there's an order to operations here. It looks like the checks are happening piecemeal. But we should probably do ALL validation upfront. Only once we're certain the project is in the expected state should any migration process start.

Reply:
That's fair, I'll change that.

Reply:
Done